### DISCUSSION TITLE
I'm not able to see the username who deleted the file in wazuh email alert.

### DISCUSSION THREAD

**Discussion by sandraimmaculate** [wazuh/wazuh#19873] [General]
Hi, Mine is Ubuntu OS i have configured wazuh to send an email alert when a file is deleted. Yea it sending the alert along with which file is deleted, but its not giving the information about the user who deleted the file (username, user-id etc). but it giving the attributes of file which contains who created the file. But i want to see who deleted the file I'm unable see that in wazuh email alert and also in wazuh dashboard alert. Do i have to change anything in ossec.conf file or in rule file?    Can you help me with this!

**Comment by mdiego92** [comment]
Hello @sandraimmaculate,

You can achieve that by enabling `whodata` in y our File Integrity Monitoring configuration.
Please follow the documentation to configure it https://documentation.wazuh.com/current/user-manual/capabilities/file-integrity/advanced-settings.html

Example for Linux distributions:

```
<syscheck>
  ...
  <directories check_all="yes" whodata="yes" report_changes="yes">/etc/hosts.allow</directories>
  ...
</syscheck>
```

Example for Windows:

```
<syscheck>
  ...
  <directories check_all="yes" whodata="yes">C:\test</directories>
  ...
</syscheck>
```

Hopefully, this resolves the issue!
Looking forward to your feedback.

**Reply by sandraimmaculate** [reply]
Thanks for replying!  
I have enabled it but even though I'm not able to see the username who deleted the file. I'm able to see the exact user name only when they login (Pam session login)!. And when they modified the file I'm not able to see the username in email alert. But I'm able see the username in wazuh dashboard. But when file deletion event happens I'm not able to see the username in dashboard and also in email alert. And yea i have installed auditd also, But when i executed this command auditctl -l | grep wazuh_fim as per the documentation i have to get the otp as: auditctl -w /etc -p wa -k wazuh_fim, But im getting -w /tmp -p wa -k wazuh_fim like this there is, no auditctl infront. idk why. Is this can be a reason? why im not getting the username 



Wazuh Notification.
2023 Oct 27 05:17:47

Received From: (agent-1) 172.31.0.0.0.->syscheck
Rule: 553 fired (level 7) -> "File deleted."
Portion of the log(s):

File '/tmp/ok' deleted
Mode: whodata

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Fri Oct 27 05:17:12 2023
 - Inode: 16120
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
 - (Audit) User name: root
 - (Audit) Audit name: ubuntu
 - (Audit) Effective name: root
 - (Audit) Group name: root
 - (Audit) Process id: 2088
 - (Audit) Process name: /usr/bin/rm
 - (Audit) Process cwd: /tmp
 - (Audit) Parent process name: /usr/bin/sudo
 - (Audit) Parent process id: 2087
 - (Audit) Parent process cwd: /tmp
 this was the email alert I'm getting, its giving root as the user name and effective username. But root is not the exact user who deleted the file!

**Reply by sandraimmaculate** [reply]
Hello can you help me with this! ...................

**Reply by mdiego92** [reply]
Hello @sandraimmaculate,

Yes, it could be related to the issue. Please set up the auditctl configuration as per the documentation:
https://documentation.wazuh.com/current/user-manual/capabilities/system-calls-monitoring/how-it-works.html

Following this, you could check the section Tracking the user running root commands in the following article, pointing exactly what you're looking for!
https://wazuh.com/blog/monitoring-root-actions-on-linux-using-auditd-and-wazuh/

Looking forward to your feedback!
Best regards.

**Comment by sandraimmaculate** [comment]
no, there is no any changes even though i have configured the auditctl still I'm getting root as the username I'm not getting exact username who deleted the file, but I'm getting the exact username when they use to modify the file add the file etc..
